### PR TITLE
Fix backticks in HTML comment

### PR DESCRIPTION
**Pull request by drwpow**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,6 +36,11 @@ From the `lib/compiler` directory, run:
 npm run build
 ```
 
+## Testing
+
+- Run all tests: `go test -v ./internal/...` (the `...` is basically a “glob”)
+- Run a specific folder of tests: `go test -v ./internal/printer`
```
Comment (drwpow) [inline README.md:42]: Unrelated to PR; just wanted to have a newcomer snippet for local testing

File: README.md [hunk 2]
```diff
@@ -46,7 +51,10 @@ import { transform } from "@astrojs/compiler";
 
 const filePath = new URL("./pages/index.astro", import.meta.url);
 const src = await fs.promises.readFile(filePath);
-const result = await transform(src, { sourcefile: filePath.href, sourcemap: 'both' });
+const result = await transform(src, {
+  sourcefile: filePath.href,
+  sourcemap: "both",
+});
```
Comment (drwpow) [inline README.md:57]: Just formatting :shrug:

File: internal/printer/print-to-js.go [hunk 3]
```diff
@@ -241,7 +241,8 @@ func render1(p *printer, n *Node, opts RenderOptions) {
 	case CommentNode:
 		p.addSourceMapping(n.Loc[0])
 		p.print("<!--")
-		p.print(n.Data)
+		backticks := regexp.MustCompile("`")
+		p.print(backticks.ReplaceAllString(n.Data, "\\`"))
```
Comment (drwpow) [inline internal/printer/print-to-js.go:245]: Actual fix was easy! Just copied from above

File: internal/printer/printer_test.go [hunk 4]
```diff
@@ -144,6 +144,16 @@ const groups = [[0, 1, 2], [3, 4, 5]];
...
 		},
+		{
+			name:   "backtick in HTML comment",
+			source: "<body><!-- `npm install astro` --></body>",
+			want: want{
+				imports:     "",
+				frontmatter: []string{},
+				styles:      []string{},
+				code:        "<html><head></head><body><!-- \\`npm install astro\\` --></body></html>",
+			},
```
Comment (drwpow) [inline internal/printer/printer_test.go:155]: Test added

File: internal/printer/print-to-js.go (hunk 3 above)
Comment (natemoo-re) [reply internal/printer/print-to-js.go:245]: Nice! Would be good to maybe extract this to an `escape` function?

**Review by natemoo-re — APPROVED** [review]
LGTM! Small nit about code organization but not blocking

**Pull request merged**